### DISCUSSION TITLE
How do you unbind a key?

### DISCUSSION THREAD

**Discussion by unrealapex** [nvim-tree/nvim-tree.lua#1619] [Q&A] [answered]
I am trying to bind `-` to toggle NvimTree but it does not work as `-` is binded to go up a directory. How can I unbind this? I have tried this but it does not work:
```lua
require("nvim-tree").setup({
  sort_by = "case_sensitive",
  view = {
    adaptive_size = true,
    mappings = {
      list = {
        { key = "u", action = "dir_up" },
-- unbind -
        { key = "-", action = "" },
      },
    },
  },
  renderer = {
    group_empty = true,
  },
  filters = {
    dotfiles = true,
  },
})

```

**Comment by gegoune** [comment, marked as answer]
Try setting it to `nil`.

**Reply by unrealapex** [reply]
Unfortunately that did not work.

**Reply by alex-courtis** [reply]
That exact unmap is [working for me](https://github.com/alex-courtis/arch/blob/9abb600dcf756e960167c3da286ad3ec156b8823/config/nvim/lua/amc/nvim-tree.lua). We even have the same `u` binding...

Perhaps there is an issue around configuration setup.

Please:
* try my config
* enable diagnostic logging for config: `:help nvim-tree.log`
```lua
    log = {
      enable = true,
      truncate = true,
      types = {
        config = true,
```

**Reply by unrealapex** [reply]
I realized what I did wrong. I tried binding `-` before I unset it from NvimTree.
code before:
```lua
local bufopts = { noremap=true, silent=true, buffer=bufnr }
-- trying to set the keymap before preventing NvimTree from overriding
vim.keymap.set('n', '-', ':NvimTreeToggle<CR>', bufopts)

        { key = "-", action = nil },

```
after:
```lua
        { key = "-", action = nil },

local bufopts = { noremap=true, silent=true, buffer=bufnr }
-- setting the keymap after preventing NvimTree from overriding it
vim.keymap.set('n', '-', ':NvimTreeToggle<CR>', bufopts)
```
